doc/rst/usingchapel/debugging/sanitizers.rst

```diff
@@ -0,0 +1,109 @@
+.. _readme-sanitizers:
+
+==========
+Sanitizers
+==========
+
+Sanitizers are a compiler feature that support instrumenting programs to do
+dynamic analysis to catch many classes of bugs at runtime.
+
+AddressSanitizer (ASan) is a fast memory error detector. It consists of a
+compiler instrumentation module and a run-time library. ASan is similar to
+valgrind in that it can help identify memory errors. ASan is much faster than
+valgrind, but does require recompilation. Note that only GCC and Clang support
+sanitizers.
+
+Building the compiler
+---------------------
+
+The following chplenv variables are required to build the Chapel compiler with
+sanitizers enabled:
+
+.. code-block:: bash
+
+   export CHPL_TARGET_MEM=cstdlib
+   export CHPL_TARGET_COMPILER=gnu # or CHPL_TARGET_COMPILER=clang
+   export CHPL_SANITIZE_EXE=address
+
+Running the compiled program
+----------------------------
+
+After building a compiler configured like the above, you can build and run a
+Chapel program with AddressSanitizer enabled by setting the following
+environment variables:
+
+.. code-block:: bash
+
+   chpl <program.chpl>
```

Comment:
Mild preference for `chpl hello.chpl` and `./hello` instead of using special characters like this. I have seen people make mistakes thinking that they mean something else.